tool-execution/pmd_setup.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+## This script sets up PMD on an Ubuntu 20.04 or MacOS 11 machine.
+
+###########
+## SETUP ##
+###########
+
+## checks if the environment variable pmd is set
+if command -v pmd &> /dev/null ; then
+    echo "pmd already setup on this machine."
+    exit 2  
+fi
+
+PMD_DIR=../pmd
+
+## create pmd dir if not exists
+mkdir -p ${PMD_DIR}
+
+## navigate to pmd directory
+cd ${PMD_DIR} || exit
+
+## Check OS version
+OS=$(uname -a | grep -o '\w*' | head -1)
+
+if [ "$OS" == "Darwin" ]; then
+    echo "MacOS detected"
+    echo "Downloading PMD for MacOS..."
+elif [ "$OS" == "Linux" ]; then
+    echo "Ubuntu detected"
```

Comment:
This is not actually detecting ubuntu, but any linux distro. I'd change this to "Linux".